pkg/handler/enabled_flag_validation.go

```diff
@@ -0,0 +1,201 @@
+			return errors.New("Argument must be a string")
+		}
+	}
+
+	return nil
+}
+
+// RegexMatch determines whether key1 matches the pattern of key2 in regular expression.
+func RegexMatch(key1 string, key2 string) (bool, error) {
+	res, err := regexp.MatchString(key2, key1)
+	if err != nil {
+		return false, err
+	}
+	return res, nil
+}
+
+// RegexMatchFunc is a govaluate function for matching regex,
+// it returns true if arg1 matches the regex pattern of arg2.
+func RegexMatchFunc(args ...interface{}) (interface{}, error) {
+	if err := validateVariadicArgs(2, args...); err != nil {
+		return false, fmt.Errorf("%s: %s", "regexMatch", err)
+	}
+
+	name1 := args[0].(string)
+	name2 := args[1].(string)
+
+	result, err := RegexMatch(name1, name2)
+	if err != nil {
+		return false, err
+	}
+
+	return result, nil
+}
+
+/* end 'with thanks to' */
+
+// THIS got weird when the params were in the other order (e.g. array, regex)
+// in this case it flattened the array into the args, e.g.
+// Any([1,2], /\d+/)
+// args = 1,2,/\d+/
+// putting it in regex, array order does not flatten it
+// args = /\d+/, [1,2]
+//
+// Govaluate function to evaluate an embedded govaluate expression (the first argument, a string)
+// for an array of 'items' (which should be a map[string]interface{}),
+// and if the subexpression returns true for any of the sub items
+// then the Any expression also returns true. It will early out once
+// it finds a single true result.
+func Any(args ...interface{}) (interface{}, error) {
+	if len(args) < 2 {
+		return false, fmt.Errorf("Expected 2+ arguments, but got %d", len(args))
+	}
+
+	function, ok := args[0].(string)
+	if !ok {
+		return false, fmt.Errorf("arg 0 is not a string!")
+	}
+
+	arr, ok := args[1].([]interface{})
+	if !ok {
+		return false, fmt.Errorf("arg 1 is not a slice! ***%v*** %v", args, reflect.TypeOf(args[1]))
+	}
+
+	functions := map[string]govaluate.ExpressionFunction{
+		"regexMatch": RegexMatchFunc,
+	}
+	expression, err := govaluate.NewEvaluableExpressionWithFunctions(function, functions)
+	if err != nil {
+		return false, err
+	}
+
+	found := false
+	for _, item := range arr {
+		params, ok := item.(map[string]interface{})
+		if !ok {
+			return false, fmt.Errorf("item is not a map %v", item)
+		}
+
+		result, err := expression.Evaluate(params)
+		if err != nil {
+			return false, err
+		}
+
+		if boolResult, ok := result.(bool); ok {
+			if boolResult {
+				found = true
+				break
+			}
+		}
+	}
+	return found, nil
+}
+
+func getEvaluationRules() ([]*govaluate.EvaluableExpression, error) {
+	createEvaluationRule.Do(func() {
+		for _, structExpressionString := range config.Config.EnabledFlagValidationRules {
+			functions := map[string]govaluate.ExpressionFunction{
+				"regexMatch": RegexMatchFunc,
+				"any":        Any,
+			}
+
+			expression, err := govaluate.NewEvaluableExpressionWithFunctions(structExpressionString, functions)
+			if expression == nil || err != nil {
+				evaluationRulesSetupError = err
```

Comment:
Looks like this will exit early if we encounter any error without fully building our evaluation rule set? Should we log an error of some kind or at least `return nil, err` to propagate whatever failed up?